payu/models/model.py

```diff
@@ -286,6 +286,12 @@ def setup(self):
             # If have exe manifest this implies exe reproduce is True. Do not
             # want to overwrite exe manifest in this case
             if not self.expt.manifest.have_manifest['exe']:
+                # Check whether executable path exists
+                if not os.path.isfile(self.exec_path):
```

Comment:
I wonder if we're going to the trouble of checking, if we shouldn't also check if it is executable. I've tested and it gives the same error when the executable doesn't have the executable permissions:
```
mpirun was unable to launch the specified application as it could not access
or execute an executable:
```

`os.access` can be used to test for executable status

https://docs.python.org/3/library/os.html#os.access

So 
```python
os.access(self.exec_path, os.X_OK)
```
should do the trick. While you can do the existence check and the executable check in one step with `os.access` the message to the user should be different for the two cases, so probably best to check them separately, and sequentially.

Reply:
Thanks, I'll add that in!

Reply:
I've added it in and then most of the tests broke because all the test exe files weren't executable. I changed the permission of the test files in the test code to be executable, even though their content is random bytes so any attempt to actually execute the files would result in an error..

Reply:
Failing tests that are fixed are the best kind of tests! :)